app/src/main/java/com/keylesspalace/tusky/util/NotificationHelper.java

```diff
@@ -104,75 +133,145 @@ public static void make(final Context context, Notification body, AccountEntity
+        try {
+            accountAvatar = Picasso.with(context)
+                    .load(body.getAccount().getAvatar())
+                    .transform(new RoundedTransformation(20))
+                    .get();
+        } catch (IOException e) {
+            Log.d(TAG, "error loading account avatar", e);
+            accountAvatar = BitmapFactory.decodeResource(context.getResources(), R.drawable.avatar_default);
+        }
+
+        builder.setLargeIcon(accountAvatar);
+
+        // Reply to mention action; RemoteInput is available from KitKat Watch, but buttons are available from Nougat
+        if (body.getType() == Notification.Type.MENTION
+                && android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            RemoteInput replyRemoteInput = new RemoteInput.Builder(KEY_REPLY)
+                    .setLabel(context.getString(R.string.label_quick_reply))
+                    .build();
+
+            PendingIntent replyPendingIntent = getStatusReplyIntent(context, body, account);
+
+            NotificationCompat.Action replyAction =
+                    new NotificationCompat.Action.Builder(R.drawable.ic_reply_24dp,
+                            context.getString(R.string.action_quick_reply), replyPendingIntent)
+                            .addRemoteInput(replyRemoteInput)
+                            .build();
+
+            builder.addAction(replyAction);
+        }
+
+        builder.setSubText(account.getFullName());
+        builder.setVisibility(NotificationCompat.VISIBILITY_PRIVATE);
+        builder.setCategory(NotificationCompat.CATEGORY_SOCIAL);
+        builder.setOnlyAlertOnce(true);
+
+        // Summary
+        // =======
+        final NotificationCompat.Builder summaryBuilder = newNotification(context, body, account, true);
+
+        if (currentNotifications.length() != 1) {
+            try {
+                String title = context.getString(R.string.notification_title_summary, currentNotifications.length());
+                String text = joinNames(context, currentNotifications);
+                summaryBuilder.setContentTitle(title)
+                        .setContentText(text);
+            } catch (JSONException e) {
+                Log.d(TAG, Log.getStackTraceString(e));
+            }
+        }
+
+        summaryBuilder.setSubText(account.getFullName());
+        summaryBuilder.setVisibility(NotificationCompat.VISIBILITY_PRIVATE);
+        summaryBuilder.setCategory(NotificationCompat.CATEGORY_SOCIAL);
+        summaryBuilder.setOnlyAlertOnce(true);
+        summaryBuilder.setGroupSummary(true);
+
+        NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
+
+        //noinspection ConstantConditions
+        notificationManager.notify(Integer.valueOf(body.getId()), builder.build());
+        if (currentNotifications.length() == 1) {
+            notificationManager.notify((int) account.getId(), builder.setGroupSummary(true).build());
```

Comment:
Is it safe?..

Reply:
I mean casting to int

Reply:
It's okay, because row identifiers in database aren't random, but incremented by one for each new account, and I guess that no one would exceed the "limit" of 65535 accounts. :wink:

Reply:
Ah, it's a local id, thx, I think it's perfectly legal